pkg/cmd/server/api/validation/oauth.go

```diff
@@ -162,6 +166,24 @@ func ValidateLDAPIdentityProvider(provider *api.LDAPPasswordIdentityProvider) Va
 	return validationResults
 }
 
+// RemoteConnection fields validated separately -- this is for keystone-specific validation
+func ValidateKeystoneIdentityProvider(provider *api.KeystonePasswordIdentityProvider, identityProvider api.IdentityProvider) ValidationResults {
+	validationResults := ValidationResults{}
+	validationResults.AddErrors(ValidateRemoteConnectionInfo(provider.RemoteConnectionInfo)...)
+
+	providerURL, err := url.Parse(provider.RemoteConnectionInfo.URL)
+	if err == nil {
+		if providerURL.Scheme != "https" {
+			validationResults.AddWarnings(fielderrors.NewFieldInvalid("url", provider.RemoteConnectionInfo.URL, "Auth URL should be secure and start with https"))
+		}
+	}
```

Comment:
So I looked into this. What was needed was similar to what you have there:
```
       _, urlErrs := ValidateSecureURL(provider.RemoteConnectionInfo.URL, "url")
       validationResults.AddWarnings(urlErrs...)
```
The only issue is while it acts as a warning (and not a server-stopping error), the log message implies a real error:
```
W1014 15:16:28.928500   28181 start_master.go:278] oauthConfig.identityProvider[0].provider.url: invalid value 'http://192.168.0.198:5000/v3', Details: must use https scheme
```
With the custom validation we avoid the inaccurate message. I could refactor ValidateSecureURL to take a 'warning' vs. 'error' boolean arg, but that might be more confusing than just creating a WarnOnInsecureURL method, but with only one caller of it, I don't know if that's any better than what I already have above.